carbonfive/spraygun-react-ts
# Upgrade to React 18

Problem
=======

React 18 has been out for a while now, let's upgrade!

Solution
========

- Upgrade `react`, `react-dom`, `@testing-library/react` to the latest versions
- Use the new `createRoot` technique for rendering the app, as explained in the [React 18 upgrade guide](https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html)
- Run `yarn upgrade` to get the latest semver-compatible versions of all our indirect dependencies

Steps to Verify
---------------

You may need to delete `node_modules` before running `yarn install`.

1. Run `yarn install`
2. Run `yarn lint` to confirm linters work
3. Run `yarn test --watchAll` to confirm tests pass
4. Run `yarn start` confirm the welcome screen loads in the browser
